README.md

```diff
@@ -1,2 +1,86 @@
-# flexvolume-spring-boot-starter
-Spring boot starter to read application configuration from Azure Keyvaults mounted as flexvolumes
+# HMCTS properties volume library
+
+[![Build Status](https://travis-ci.com/hmcts/properties-volume-spring-boot-starter.svg?branch=master)](https://travis-ci.com/hmcts/properties-volume-spring-boot-starter)
+[ ![Download](https://api.bintray.com/packages/hmcts/hmcts-maven/properties-volume-spring-boot-starter/images/download.svg) ](https://bintray.com/hmcts/hmcts-maven/properties-volume-spring-boot-starter/_latestVersion)
+
+This is a Spring Boot _starter_ library to read application configuration properties from files using the file name as key 
+and the file content as value. This can be used in applications that have Azure Keyvaults mounted as flexvolumes, but there 
+are probably other valid uses.
+
+
+## Usage
+Include the module into the dependency list.
+```groovy
+ compile 'uk.gov.hmcts.reform:properties-volume-spring-boot-starter:x.x.x'
+```
+
+To define which directories or files contain Azure Keyvault entries update *bootstrap.yaml* to define the related paths.
+For instance:
+```yaml
+spring:
+  application:
+    name: properties-volume-example
+  cloud:
+    propertiesvolume:
+      enabled: true
+      prefixed: true
+      paths: /mnt/secrets/this,/mnt/secrets/that,/mnt/secrets/other
```

Comment:
would it be possible for you to link a PR here showing this being used on a java app?
I would like to see two things,
1. it working
2. if you aren't using flexvolume in the app it also works, so that we can merge code to master even though the target environment is still and App service, i.e. it works for AKS, local, and App services

Reply:
@timja: Sure. Which applications would you like to target for each type you mention?
@KevinJCross: can you please check if I interpreted your requests correctly?

Reply:
I had a look for one of ours, but I don't think we have any that have secrets that have got charts written,
this could be a good one:
https://github.com/hmcts/cmc-claim-store/blob/master/charts/cmc-claim-store/values.template.yaml
 
otherwise plum-recipes would be the other option

Reply:
@timja isn't that only aks (i.e. helm chart)? what about local and app service?

Reply:
for app service you would have to deploy one of them to an app service (probably sandbox or saat) with the same code change as in the PR for the AKS one

Local is the same as app service really I think

Reply:
I would suggest just putting just `secrets` into the paths parameter (simple easy and covers our use case) and let the code find the directories and use them as a prefix to fill out the properties.
i.e. from above `paths = 'secrets' `

This way the property names would be secrets.this.aProperty or `secrets.that.a-property` using the folder/file structure to name the property e.g secrets/that/a_prop would be secrets.that.a_prop. It should be simple to File.walk and represent the folder and file structure in a property :smile: and it makes the module configuration super simple.

From my perspective representing the properties like this is basically  like a folder/file representation of the yaml or json object structure and is a very versatile way of representing/reading the properties.